### PR TITLE
Upgrade ember version

### PR DESCRIPTION
**Pull request by adrigzr**

### CODE REVIEW

File: addon/services/state.js [hunk 1]
```diff
@@ -121,7 +122,8 @@ export default Service.extend(Evented, {
 
 			window.addEventListener('popstate', this.get('_popstateDidChangeBinding'));
 
-			getOwner(this).lookup('router:main').on('willTransition', this.get('_transitionWillChangeBinding'));
+			getOwner(this).lookup('service:router').on('routeWillChange', this.get('_routeWillChangeBinding'));
```
Comment (svsalvador) [inline addon/services/state.js]: Nice

File: tests/acceptance/services/before-model-test.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	setupApplicationTest(hooks);
+
+	hooks.beforeEach(async function() {
+		await visit('/');
+
+		this.service = this.owner.lookup('service:state');
+	});
+
+	test('it checks replace in beforeModel', async function(assert) {
+		const pointer = this.service.get('current.index');
```
Comment (svsalvador) [inline tests/acceptance/services/before-model-test.js]: This property could be in `beforeEach` hook. It's the same code in every test, doesn´t it?

Comment (adrigzr) [reply tests/acceptance/services/before-model-test.js]: Yep, but I think it is part of the _Arrange_ part of the test itself, so it is good to be here.

File: tests/acceptance/services/state-test.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+		await visit('/foo');
+
+		await replace('/bar');
+
+		assert.equal(this.service.get('current.index'), pointer + 1, 'current index points to history index');
+		assert.equal(this.service.get('previous.index'), pointer + 0, 'previous index points to history before current index');
+		assert.equal(this.service.get('last.index'), pointer + 0, 'last index points to last state index');
+	});
+
+	skip('it has no previous pointer when replacing first route', async function(assert) {
```
Comment (svsalvador) [inline tests/acceptance/services/state-test.js]: These skips?

File: tests/test-helper.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import Application from '../app';
+import config from '../config/environment';
 import { setApplication } from '@ember/test-helpers';
-import './helpers/back';
-import './helpers/replace';
+// import './helpers/back';
```
Comment (svsalvador) [inline tests/test-helper.js]: Remove this code, please.

File: tests/acceptance/services/state-test.js (hunk 3 above)
Comment (adrigzr) [reply tests/acceptance/services/state-test.js]: No longer works with current testing solution =/

**Pull request merged**